src/poetry/repositories/http.py

```diff
@@ -43,7 +43,7 @@ def __init__(
         super().__init__(name, disable_cache)
         self._url = url
         self._authenticator = Authenticator(
-            config=config or Config(use_environment=True),
+            config=config,
```

Comment:
Ah yeah; this happened because I had to allow the default authenticator to delay the initialisation for the `Config` class. It is sort of related a the default value of `Authenticator`'s `config` param changed.